posts/what-is-machine-learning.md

```diff
@@ -0,0 +1,22 @@
+
+In unsupervised learning the machine learning model determines the output based on the data that is given. Then future data can be placed into the appropriate output. A common use case for this would be clustering of like information. The clusters are not predetermined but decided by the machine learning algorithm. One use case for a clustering algorithm is to group similar movies together. A movie database would provide details about the movie and then cluster similar movies together. By allowing the model to be created without dictating what the clusters should be will allow the machine to determine what features cause movies to be similar to one another. This will allow different suggestions that typically might not have been thought of before.
+
+Both supervised and unsupervised learning need to be programmed to get an output model using an algorithm. So what is this algorithm and how do you determine which one to use? The algorithms are mathematical equations used to build the model. Since they are mathematical equations any text needs to be converted to numbers. Converting the text to number is called feature extraction. There are different ways of doing this but one of the most popular ones is the TF-IDF which stands for “term frequency-inverse document frequency”. In simple terms this means in a collection of documents we determine how important a word is by calculating how many times that word appears in the documents and how many documents the word appears in. For example in machine learning algorithms for determining if an email is spam a word can be identified as a “spam word”. If a word appears in the majority of the emails that were marked as spam and appears infrequently in the emails that were marked as not spam, the word can be used to identify emails that are spam. Now that the features have been extracted, the correct algorithm needs to be picked. There are multiple different algorithms that can fulfill a specific purpose so multiple models should be tested to get the best model for your data. First, to choose the correct algorithm, you must know if you are going to use supervised or unsupervised learning. There are different algorithms for each one. Next, with supervised learning based on your use case you need to decide between classification and regression algorithms. Classification algorithms as previously discussed are for labelling data in a specific way such as type of document or spam emails. Regression algorithms are used when the output of the data are continuous values. A simple use case for regression would be predicting salaries based on years of experience. 
```

Comment:
TF-IDF is for blobs of text. Maybe also worth mentioning that we can turn textual categories into numbers.